packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1014,6 +1020,45 @@ describe("BnsConnection", () => {
 
       connection.disconnect();
     });
+
+    it("reports DeliverTx errors for search by ID", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId } = await userProfileWithFaucet(chainId);
+      // this will never have tokens, but can try to sign
+      const brokeIdentity = await profile.createIdentity(mainWalletId, chainId, HdPaths.simpleAddress(1234));
+
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        creator: brokeIdentity,
+        recipient: await randomBnsAddress(),
+        memo: "Sending from empty",
+        amount: defaultAmount,
+      };
+
+      const nonce = await connection.getNonce({ pubkey: brokeIdentity.pubkey });
+      const signed = await profile.signTransaction(mainWalletId, brokeIdentity, sendTx, bnsCodec, nonce);
+      const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+      const transactionIdToSearch = response.transactionId;
+      await response.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
```

Comment:
Same point with waitForCondition (`transactionProcessed`)

Reply:
What's the issue here? This is the default statement for _wait for block_